api/include/opentelemetry/common/spin_lock_mutex.h

```diff
@@ -68,8 +68,10 @@ class SpinLockMutex
 #  else
     __builtin_ia32_pause();
 #  endif
-#elif defined(__arm__)
-    __asm__ volatile("yield" ::: "memory");
+#elif defined(__armel__) || defined(__ARMEL__)
+    asm volatile("nop" ::: "memory");
```

Comment:
Is there any difference between `asm volatile() and `__asm__ __volatile__()`?

Reply:
AFAIU, there is no difference between them, and it mostly depends on the projects conventions and style. Given that there are other occurrences of  `__asm__ volatile(...)`, and zero occurrences of  `asm volatile(...)` in other parts of the code, I might change it if you want.